src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -1080,16 +1080,15 @@ private void GetConnection()
 
         private void Abort()
         {
-            // Try-catch commented out to identify exception that is causing
-            // https://github.com/dotnet/corefx/issues/40711.
-
-            // try
-            // {
+            try
+            {
                 _transport.Abort();
```

Comment:
Maybe we should put this in `SmtpConnection.Abort`, and omit this try/catch?
This is the stacktrace for the exception:
```
Object name: 'SafeHandle'.
   at System.Runtime.InteropServices.SafeHandle.DangerousAddRef(Boolean& success)
   at System.Net.Sockets.SocketPal.SetLingerOption(SafeSocketHandle handle, LingerOption optionValue)
   at System.Net.Sockets.Socket.SetLingerOption(LingerOption lref)
   at System.Net.Sockets.Socket.SetSocketOption(SocketOptionLevel optionLevel, SocketOptionName optionName, Object optionValue)
   at System.Net.Sockets.Socket.set_LingerState(LingerOption value)
   at System.Net.Sockets.TcpClient.set_LingerState(LingerOption value)
   at System.Net.Mail.SmtpConnection.Abort()
   at System.Net.Mail.SmtpTransport.Abort()
   at System.Net.Mail.SmtpClient.Abort()
   at System.Net.Mail.SmtpClient.TimeOutCallback(Object state)
   at System.Threading.TimerQueueTimer.<>c.<.cctor>b__23_0(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.TimerQueueTimer.CallCallback(Boolean isThreadPool)
   at System.Threading.TimerQueueTimer.Fire(Boolean isThreadPool)
   at System.Threading.TimerQueue.FireNextTimers()
   at System.Threading.TimerQueue.System.Threading.IThreadPoolWorkItem.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
   at System.Threading.Thread.StartCallback()
```

Reply:
Changed.